homeassistant/components/rainbird/number.py

```diff
@@ -58,4 +61,7 @@ def native_value(self) -> float | None:
 
     async def async_set_native_value(self, value: float) -> None:
         """Update the current value."""
-        await self.coordinator.controller.set_rain_delay(value)
+        try:
+            await self.coordinator.controller.set_rain_delay(value)
+        except RainbirdApiException as err:
+            raise HomeAssistantError() from err
```

Comment:
We should pass an error message here, that is displayed to the user.

Looking upstream, the message raised is rather technical; this should have a end-user consumable message.

../Frenck

Reply:
Thank you for the feedback. My thought is to improve the underlying messages (move some details to logs of an appropriate level, and change the exception hierarchy to remove "internal" super technical errors to not inherit the main exceptions). I think this may be simpler rater than re-interpreting all the exceptions in home assistant. https://github.com/home-assistant/core/pull/98271

Will revisit this after bumping the llibrary.

Reply:
I'm not fully in agreement, the context here is different.

One of the future ideas is to add translations to them, that would not work either.

Reply:
Ok, yes, translations makes sense. To the point of user friendly error messages, I'd rather not invent my own unique way to say 'there was an error talking to the device" so I welcome this.

Either way the other PR to bump improves the error hierarchy but it doesn't need to block this.

I'll work on incorporating the mirrored error hierarchy here too. The nice thing about doing this in a single place for all API calls, so will think about how to reduce copies code, otherwise will copy across all three places and test the variations on all 3 places.